contracts/modules/Swap.sol

```diff
@@ -393,6 +493,8 @@ contract Swap is BaseLogic {
     }
 
     function finalizeSwap(SwapCache memory swap) private {
+        Utils.safeApprove(swap.assetCacheIn.underlying, uniswapRouter, 0);
```

Comment:
`finalizeSwap` is also called by `swap1Inch` right? In this case it was `oneInch` that was granted the approval, not `uniswapRouter`.